libs/langchain-google-common/src/experimental/utils/media_core.ts

```diff
@@ -0,0 +1,674 @@
+import { v1, v4 } from "uuid"; // FIXME - it is importing the wrong uuid, so v6 and v7 aren't implemented
+import { BaseStore } from "@langchain/core/stores";
+import {
+  Serializable,
+  Serialized,
+  SerializedConstructor,
+} from "@langchain/core/load/serializable";
+
+export type MediaBlobData = {
+  value: string; // In Base64 encoding
+  type: string; // The mime type and possibly encoding
+};
+
+export interface MediaBlobParameters {
+  data?: MediaBlobData;
+
+  metadata?: Record<string, unknown>;
+
+  path?: string;
+}
+
+function bytesToString(dataArray: Uint8Array): string {
+  // Need to handle the array in smaller chunks to deal with stack size limits
+  let ret = "";
+  const chunkSize = 102400;
+  for (let i = 0; i < dataArray.length; i += chunkSize) {
+    const chunk = dataArray.subarray(i, i + chunkSize);
+    ret += String.fromCharCode(...chunk);
+  }
+
+  return ret;
+}
+
+/**
+ * Represents a chunk of data that can be identified by the path where the
+ * data is (or will be) located, along with optional metadata about the data.
+ */
+export class MediaBlob extends Serializable implements MediaBlobParameters {
+  lc_serializable = true;
+
+  lc_namespace = ["langchain", "google-common"]; // FIXME - What should this be? And why?
+
+  data?: MediaBlobData;
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  metadata?: Record<string, any>;
+
+  path?: string;
+
+  constructor(params?: MediaBlobParameters | Serialized) {
+    const sparams = params as SerializedConstructor;
+    super(sparams?.kwargs);
```

Comment:
This is a bit odd - you can just pass `params`

If it extends `Serializable`, any object params you pass will automatically be serialized

Reply:
Why the cast and `.kwargs`? Should probably be a type guard if anything?

But also why `| Serialized`?